### PR TITLE
add sphinx doc using auto api

### LINKED ISSUE #26: Convert docstrings into documentation
We've done a great job so far of documenting classes and functions. The next step is to convert this into documentation that can be hosted on something like [Read the Docs](https://readthedocs.org/).

### PR DESCRIPTION
**Pull request by bicycleman15**
Fixes #26 
Added documentation using sphinx-autoapi

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Configuration file for the Sphinx documentation builder.
```
Comment (AlanAboudib) [inline docs/source/conf.py]: Please apply `black` to this file in order to pass CI tests

**Comment by bicycleman15** [comment]
To test it @MarcioPorto , first run
`pip install sphinx sphinx_rtd_theme sphinx-autoapi`
Then inside the `docs` folder, run
`make html`
docs then will be inside the `_build` folder, you navigate through them by opening `docs/_build/html/index.html`

**Comment by AlanAboudib** [comment]
@MarcioPorto any updates on this PR?

**Comment by MarcioPorto** [comment]
@AlanAboudib I have been adding a few final touches locally. I am hoping to wrap it up this weekend.

**Comment by Nilanshrajput** [comment]
done in #164.

**Pull request closed without merging**